lib/cloudant.js

```diff
@@ -354,14 +356,20 @@ Cloudant.prototype.buildSort = function(mo, model, order) {
  * @param {Object} options The options object
  * @param [cb] The cb function
  */
-Cloudant.prototype.destroy = function destroy(model, id, rev, options, cb) {
-  debug('Cloudant.prototype.destroy %j %j %j', model, id, rev, options);
-  var self = this;
-  var mo = self.selectModel(model);
-  id = id.toString();
-  mo.db.destroy(id, rev, function(err, result) {
-    if (err) return cb(err, null);
-    cb(null, {id: id, rev: rev, count: 1});
+Cloudant.prototype.destroy = function destroy(model, id, options, cb) {
+  var mo = this.selectModel(model);
+  this.all(model, {where: {id: id}}, {raw: true}, function(err, doc) {
+    if (doc.length > 1) cb(new Error(
+      'instance method destroy tries to delete more than one item!'));
+    else if (doc.length === 1) {
+      mo.db.destroy(doc[0]._id, doc[0]._rev, function(err, result) {
+        debug('Cloudant.prototype.destroy db.destroy %j %j', err, result);
+        if (err) return cb(err);
+        cb(err, result && result.ok ? 1 : 0);
```

Comment:
Are we changing the return of `destroy`? Is the current way you specified as the default way for other connectors?